code/game/mecha/mech_fabricator.dm

```diff
@@ -475,7 +461,15 @@
 	component_parts += new /obj/item/stock_parts/micro_laser(null)
 	component_parts += new /obj/item/stack/sheet/glass(null)
 	RefreshParts()
+	files = new /datum/research(src) //Setup the research data holder.
+	matching_designs = list()
+
+/obj/machinery/mecha_part_fabricator/spacepod/interact(mob/user)
+	. = ..()

```

Comment:
Do you need to write this? If you remove this proc does this not automatically happen? (I'm don't know)